pkg/trace/obfuscate/sql_test.go

```diff
@@ -466,65 +750,96 @@ func TestConsumerError(t *testing.T) {
+		{
+			"SELECT name, `age}` FROM profile",
+			`at position 17: literal identifiers must end in "` + "`" + `", got "}" (125)`,
+		},
 
-	_, err = obfuscateSQLString("SELECT name, `age}` FROM profile")
-	assert.Error(err)
-	assert.Equal(`at position 17: literal identifiers must end in "`+"`"+`", got "}" (125)`, err.Error())
+		{
+			"SELECT %(asd)| FROM profile",
+			`at position 13: invalid character after variable identifier: "|" (124)`,
+		},
 
-	_, err = obfuscateSQLString("SELECT %(asd)| FROM profile")
-	assert.Error(err)
-	assert.Equal(`at position 13: invalid character after variable identifier: "|" (124)`, err.Error())
+		{
+			"USING $A FROM users",
+			`at position 7: prepared statements must start with digits, got "A" (65)`,
+		},
 
-	_, err = obfuscateSQLString("USING $A FROM users")
-	assert.Error(err)
-	assert.Equal(`at position 7: prepared statements must start with digits, got "A" (65)`, err.Error())
+		{
+			"USING $09 SELECT",
+			`at position 9: invalid number`,
+		},
 
-	_, err = obfuscateSQLString("USING $09 SELECT")
-	assert.Error(err)
-	assert.Equal(`at position 9: invalid number`, err.Error())
+		{
+			"INSERT VALUES (1, 2) INTO {ABC",
+			`at position 30: unexpected EOF in escape sequence`,
+		},
 
-	_, err = obfuscateSQLString("INSERT VALUES (1, 2) INTO {ABC")
-	assert.Error(err)
-	assert.Equal(`at position 30: unexpected EOF in escape sequence`, err.Error())
+		{
+			"SELECT one, :2two FROM profile",
+			`at position 13: bind variables should start with letters, got "2" (50)`,
+		},
 
-	_, err = obfuscateSQLString("SELECT one, :2two FROM profile")
-	assert.Error(err)
-	assert.Equal(`at position 13: bind variables should start with letters, got "2" (50)`, err.Error())
+		{
+			"SELECT age FROM profile WHERE name='John \\",
+			`at position 43: unexpected EOF in string`,
+		},
 
-	_, err = obfuscateSQLString("SELECT age FROM profile WHERE name='John \\")
-	assert.Error(err)
-	assert.Equal(`at position 42: unexpected EOF after escape character in string`, err.Error())
+		{
+			"SELECT age FROM profile WHERE name='John",
+			`at position 41: unexpected EOF in string`,
+		},
+
+		{
+			"/* abcd",
+			`at position 7: unexpected EOF in comment`,
+		},
 
-	_, err = obfuscateSQLString("SELECT age FROM profile WHERE name='John")
-	assert.Error(err)
-	assert.Equal(`at position 41: unexpected EOF in string`, err.Error())
+		// using mixed cases of backslash escaping the single quote
+		{
+			"SELECT age FROM profile WHERE name='John\\' and place='John\\'s House'",
+			`at position 59: unexpected byte 92`,
+		},
 
-	_, err = obfuscateSQLString("/* abcd")
-	assert.Error(err)
-	assert.Equal(`at position 7: unexpected EOF in comment`, err.Error())
+		{
+			"SELECT age FROM profile WHERE place='John\\'s House' and name='John\\'",
+			`at position 69: unexpected EOF in string`,
+		},
+	}
+	for i, tc := range cases {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
```

Comment:
Trick: you can simply use `t.Run("", ...` and drop the `i`. The `go` tool will number them for you.

Reply:
Please do. Copying is the best way to continue suboptimal practices :) Hopefully the next person that copies this, will copy your version and not the other one. I'm sorry that the repo is not as consistent as you'd like to. I can actually recognise myself in this code and you've probably copied it from somewhere I did, before learning the above shortcut. There's really no reason to have an index and use `strconv` package when go can just do it for you.